ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/ExecutionEngineJsonRpcMethods.java

```diff
@@ -74,56 +80,87 @@ protected String getApiGroup() {
+      executionEngineApisSupported.addAll(
+          Arrays.asList(
+              new EngineGetPayloadV1(
+                  consensusEngineServer,
+                  protocolContext,
+                  mergeCoordinator.get(),
+                  blockResultFactory,
+                  engineQosTimer,
+                  protocolSchedule),
+              new EngineGetPayloadV2(
+                  consensusEngineServer,
+                  protocolContext,
+                  mergeCoordinator.get(),
+                  blockResultFactory,
+                  engineQosTimer,
+                  protocolSchedule),
+              new EngineNewPayloadV1(
+                  consensusEngineServer,
+                  protocolSchedule,
+                  protocolContext,
+                  mergeCoordinator.get(),
+                  ethPeers,
+                  engineQosTimer),
+              new EngineNewPayloadV2(
+                  consensusEngineServer,
+                  protocolSchedule,
+                  protocolContext,
+                  mergeCoordinator.get(),
+                  ethPeers,
+                  engineQosTimer),
+              new EngineNewPayloadV3(
+                  consensusEngineServer,
+                  protocolSchedule,
+                  protocolContext,
+                  mergeCoordinator.get(),
+                  ethPeers,
+                  engineQosTimer),
+              new EngineForkchoiceUpdatedV1(
+                  consensusEngineServer,
+                  protocolSchedule,
+                  protocolContext,
+                  mergeCoordinator.get(),
+                  engineQosTimer),
+              new EngineForkchoiceUpdatedV2(
+                  consensusEngineServer,
+                  protocolSchedule,
+                  protocolContext,
+                  mergeCoordinator.get(),
+                  engineQosTimer),
+              new EngineGetBlobsBundleV1(
+                  consensusEngineServer,
+                  protocolContext,
+                  mergeCoordinator.get(),
+                  new BlockResultFactory(),
+                  engineQosTimer,
+                  protocolSchedule),
+              new EngineExchangeTransitionConfiguration(
+                  consensusEngineServer, protocolContext, engineQosTimer),
+              new EngineGetPayloadBodiesByHashV1(
+                  consensusEngineServer, protocolContext, blockResultFactory, engineQosTimer),
+              new EngineGetPayloadBodiesByRangeV1(
+                  consensusEngineServer, protocolContext, blockResultFactory, engineQosTimer),
+              new EngineExchangeCapabilities(
+                  consensusEngineServer, protocolContext, engineQosTimer),
+              new EnginePreparePayloadDebug(
+                  consensusEngineServer, protocolContext, engineQosTimer, mergeCoordinator.get())));
+
+      if (protocolSchedule.anyMatch(p -> p.spec().getName().equalsIgnoreCase("cancun"))) {
```

Comment:
Why does the introduction of V3 need to be triggered by Cancun? With Shanghai, some CLs started calling the next API version before the fork was triggered.

Reply:
It's more for configuration consistency than timing. If you can't hit V3 of this api, maybe you don't have cancun enabled, and you're gonna need to fix that.

Will consider removing this though, and just always enabling it since the api itself is timestamp aware.